sql/src/test/java/io/crate/planner/consumer/SemiJoinsTest.java

```diff
@@ -160,8 +160,8 @@ public void testDisabledByDefault() throws Exception {
     public void testWriteWithMultipleInClauses() throws Exception {
         SelectAnalyzedStatement stmt = executor.analyze("select * from t1 " +
                                                         "where " +
-                                                        "   x in (select * from unnest([1, 2])) " +
-                                                        "   and x not in (select 1)");
+                                                        "   x in (select col1::integer from unnest([1, 2])) " +
```

Comment:
I wonder if we should do something about this case too?
With the current logic the type origin are literals, but they're not polymorphic as in other cases

Reply:
We could. The issue here is the `SelectSymbol` doesn't get normalized as other symbols. Otherwise we could upcast from long to integer.